lib/api/controllers/authController.js

```diff
@@ -105,7 +109,40 @@ class AuthController {
    * @returns {Promise<object>}
    */
   getCurrentUser(request) {
-    return formatProcessing.formatUserForSerialization(this.kuzzle, request.context.user);
+    let formattedUser;
+
+    return formatProcessing.formatUserForSerialization(this.kuzzle, request.context.user)
+      .then(result => {
+        const
+          availableStrategies = this.kuzzle.pluginsManager.listStrategies(),
+          promises = [];
+
+        formattedUser = result;
+        
+        if (this.kuzzle.repositories.user.anonymous()._id === request.context.token.userId) {
+          return Promise.resolve([]);
+        }
+
+        for (let i = 0; i < availableStrategies.length; i++) {
```

Comment:
@scottinet why? 

To me, es6 `for .. of` is much clearer than `forEach`, does not create a closure and allows to break.
`for ( ; ; )` has the only disadvantages of being somewhat not as syntaxically clear and needs an extra variable but it would still be better than `forEach`.